keep/providers/gitlab_provider/gitlab_provider.py

```diff
@@ -0,0 +1,193 @@
+"""
+GitlabProvider is a class that implements the BaseProvider interface for GitLab updates.
+"""
+
+import dataclasses
+import urllib.parse
+
+import pydantic
+import requests
+from requests import HTTPError
+
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig, ProviderScope
+
+
+@pydantic.dataclasses.dataclass
+class GitlabProviderAuthConfig:
+    """GitLab authentication configuration."""
+
+    host: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "GitLab Host",
+            "sensitive": False,
+            "hint": "example.gitlab.com",
+        }
+    )
+
+    personal_access_token: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "GitLab Personal/Project/Group Access Tokens",
+            "sensitive": True,
+            "documentation_url": "https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html",
+        }
+    )
+
+
+class GitlabProvider(BaseProvider):
+    """Enrich alerts with GitLab tickets."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
```

Comment:
there are plenty of GitLab scopes - 
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#personal-access-token-scopes

Please understand which are relevant and check them in validate_scopes. Keep in mind some of there are "must have" and some of them are not